Henry's Upgrade

mengfoobar opened the pull request:
Completed challenge as specified on the instructions.

Cheers!

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+
+        server.on('load', (todos) => {
+            this.setState({
+                todos:todos
```
ES6 shorthand -> this.setState({ todos })

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+            this.setCheck(index)
+        });
+
+        server.on('disconnect', ()=>{
```
nice!

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+            return;
+        }
+
+        const curTodos =  List(todos);
```
Nice use of Immutable 👍

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+    }
+
+    add(todo=null){
+        const {todoInput, todos} = this.state;
```
Good code style

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+                        return (
+                            <TodoItem key={i} todo={t} index={i}
+                                      setCheck={()=>{
+                                          cacheMode ? this.setCheck(i) : server.emit('setCheck', i)
```
nice solution for handling disconnects

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+        this.setState({todoInput: e.target.value});
+    }
+
+    add(todo=null){
```
It would have been beneficial to clear the input after adding a todo

mengfoobar replied:
added :)

NicHaley replied:
nice!

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,223 @@
+
+        return (
+            <div>
+                <input id="todo-input" type="text" placeholder="Feed the cat" value={todoInput}
```
Would have been nice to handle adding a todo on Enter with `onKeyPress`

mengfoobar replied:
agreed! just added it

mengfoobar commented:
@NicHaley thanks so much for the code review! really appreciate your time and valuable feedback.
I fixed the suggestions you proposed  :)  Totally biffed it on the complete all one..., just added it on :+1: 

As for additional styles, I didn't add any more styling other than making it mobile responsive, because by "design change", I thought it meant adding an additional functional design (which is where the undo feature came from)  （´∀｀）

Again, thank you so much, always glad to get constructive criticism from a seasoned dev!

NicHaley commented:
Np Henry!

NicHaley commented on app/components/app.jsx:
```diff
@@ -0,0 +1,256 @@
+    render() {
+        const {todoInput, todos, mobile, cacheMode} = this.state;
+        const {mInputBox,mBtn} = style;
+        const btnClass = `${mobile? mBtn:""}`
```
Would have been better to handle mobile styles with CSS media queries instead of with javascript
